addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -839,7 +839,8 @@ const Wysiwyg = Widget.extend({
      * @returns {Boolean}
      */
     isDirty: function () {
-        return this._initialValue !== (this.getValue() || this.$editable.val());
+        const isDocumentDirty = this.$editable[0].ownerDocument.defaultView.$(".o_dirty").length;
+        return this._initialValue !== (this.getValue() || this.$editable.val()) && isDocumentDirty;
```

Comment:
When `this._initialValue !== (this.getValue() || this.$editable.val())` is `true` but `isDocumentDirty` is `false`, what are the differences between `this._initialValue` and `(this.getValue() || this.$editable.val())` ? There must be a difference, otherwise there would be not bug to fix here, so could you tell me what the difference is exactly ?

Reply:
@dmo-odoo There are some cases in which the DOM is changed even though we don't perform any operation.
Like, 1) If there are multiple menus available on the website, and we start editing the website some menus will convert into dropdowns.
2) Only scrolling up and down on the website will sometimes add or remove tags and attributes.
At a time like this value of this._initialvalue and (this.getValue() || this.$editable.val()) will not match. So, for those cases, I have checked isDocumentDirty.

Reply:
@Goaman I'm not very fond of the proposed fix here. Do you think we could filter out the changes mentionned by @dhba-odoo instead ? I would prefer it but in a way it might be very hard (or maybe even impossible) to list all possible cases to ignore. At least the proposed fix has the merit of being more generic, in a way. What do you think ?

Reply:
@dmo-odoo It is not the first time I see the value of the editable `(this.getValue() || this.$editable.val())` being different even without any modification. At some point, it happened with the checklist in `html_field` (I'm not sure if it still does but anyway, html_field has a custom dirty check). 
For me, we should *not* rely on `(this.getValue() || this.$editable.val())`.
Checking for the class "o_dirty" in website seems a more robust approach.